public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -0,0 +1,241 @@
+import { AGGREGATION_OPERATORS, RANGE_VEC_FUNCTIONS } from '../../../syntax';
+
+import { CompletionDataProvider } from './CompletionDataProvider';
+import { NeverCaseError } from './NeverCaseError';
+import type { Situation, Label } from './situation';
+
+export type CompletionType =
+  | 'HISTORY'
+  | 'FUNCTION'
+  | 'DURATION'
+  | 'LABEL_NAME'
+  | 'LABEL_VALUE'
+  | 'PATTERN'
+  | 'PARSER'
+  | 'LINE_FILTER'
+  | 'LINE_FORMAT';
+
+type Completion = {
+  type: CompletionType;
+  label: string;
+  insertText: string;
+  detail?: string;
+  documentation?: string;
+  triggerOnInsert?: boolean;
+  isSnippet?: boolean;
+};
+
+const LOG_COMPLETIONS: Completion[] = [
+  {
+    type: 'PATTERN',
+    label: '{}',
+    insertText: '{$0}',
+    isSnippet: true,
+    triggerOnInsert: true,
+  },
+];
+
+const AGGREGATION_COMPLETIONS: Completion[] = AGGREGATION_OPERATORS.map((f) => ({
+  type: 'FUNCTION',
+  label: f.label,
+  insertText: `${f.insertText ?? ''}($0)`, // i don't know what to do when this is nullish. it should not be.
+  isSnippet: true,
+  triggerOnInsert: true,
+  detail: f.detail,
+  documentation: f.documentation,
+}));
+
+const FUNCTION_COMPLETIONS: Completion[] = RANGE_VEC_FUNCTIONS.map((f) => ({
+  type: 'FUNCTION',
+  label: f.label,
+  insertText: `${f.insertText ?? ''}({$0}[\\$__interval])`, // i don't know what to do when this is nullish. it should not be.
+  isSnippet: true,
+  triggerOnInsert: true,
+  detail: f.detail,
+  documentation: f.documentation,
+}));
+
+const DURATION_COMPLETIONS: Completion[] = [
+  '$__interval',
+  '$__range',
+  '$__rate_interval',
```

Comment:
```suggestion
```

`$__rate_interval` is not supported for Loki.

Reply:
Removed!